src/test/java/com/snowflake/kafka/connector/streaming/iceberg/sql/PrimitiveJsonRecord.java

```diff
@@ -0,0 +1,177 @@
+package com.snowflake.kafka.connector.streaming.iceberg.sql;
+
+import static net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
+
+import com.snowflake.kafka.connector.Utils;
+import com.snowflake.kafka.connector.streaming.iceberg.sql.MetadataRecord.RecordWithMetadata;
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.annotation.JsonCreator;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.annotation.JsonProperty;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.ObjectMapper;
+import org.assertj.core.api.Assertions;
+
+public class PrimitiveJsonRecord {
+
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper().configure(FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+  private final Long idInt8;
+
+  private final Long idInt16;
+
+  private final Long idInt32;
+
+  private final Long idInt64;
+
+  private final String description;
+
+  private final Double ratingFloat32;
+
+  private final Double ratingFloat64;
+
+  private final Boolean approval;
+
+  @JsonCreator
+  public PrimitiveJsonRecord(
+      @JsonProperty("id_int8") Long idInt8,
+      @JsonProperty("id_int16") Long idInt16,
+      @JsonProperty("id_int32") Long idInt32,
+      // FIXME: there is currently some bug in Iceberg when storing int64 values
```

Comment:
Let's have some JIRA to remember about this problem.

Reply:
https://snowflakecomputing.atlassian.net/browse/SNOW-1754474